spec/unit/knife/cookbook_metadata_spec.rb

```diff
@@ -174,6 +175,15 @@
         expect(@stderr.string).to match /#{description} blah/im
       end
     end
+    it "should raise a syntax error on invalid JSON" do
+      expect(File).to receive(:exist?).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(true)
+      expect(IO).to receive(:read).with("#{@cookbook_dir}/foobar/metadata.json").
+        and_return(@invalid_json_data)
```

Comment:
see the spec/data/cookbooks directory and any spec that reads from it.  CHEF_SPEC_DATA can be used to grab that path.  `File.expand_path(File.join(CHEF_SPEC_DATA, "cookbooks")),` is the cookbooks dir.  there's already invalid metadata in that cookbooks dir, and there's tests in spec/unit/cookbook_loader_spec.rb for example.   if you modify that directory to have a cookbook with bad metadata.json it might cause other specs to blow up, but that might actually be good in that it would give better "functional-ish" coverage to not only test the metadata class but also test that the cookbook_loader uses that code and appropriately blows up.